### PR TITLE
zeyuan's todo

### PR DESCRIPTION
**Pull request by ZenoTan**
Finished balance part and defined report. Still need to debug and finish report part.

Update: report part has completed.

### CODE REVIEW

File: bench.go [hunk 1]
```diff
@@ -1,14 +1,40 @@
...
-import "time"
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/prometheus/client_golang/api"
+	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
```
Comment (lhy1024) [inline bench.go:11]: v1?

File: bench.go [hunk 2]
```diff
@@ -1,14 +1,40 @@
...
 	run() error
 	collect() error
 }
 
+type timePoint struct {
+	addTime     time.Time
+	balanceTime time.Time
+}
+
+type stats struct {
```
Comment (lhy1024) [inline bench.go:25]: will we collect the balance region operator?

File: bench.go [hunk 3]
```diff
@@ -30,8 +57,53 @@ func (s *scaleOut) run() error {
...
 func (s *scaleOut) isBalance() bool {
-	// todo @zeyuan get data from prometheus
-	return false
+	// todo get data from prometheus
+	// todo @zeyuan
+	client, err := api.NewClient(api.Config{
+		Address: s.c.prometheus,
+	})
+	if err != nil {
+		fmt.Printf("Error creating client: %v\n", err)
```
Comment (lhy1024) [inline bench.go:66]: we can replace it with log.fatal

File: bench.go [hunk 4]
```diff
@@ -60,11 +132,134 @@ func (s *scaleOut) collect() error {
 
 func (s *scaleOut) createReport() (string, error) {
 	//todo @zeyuan
-	return "", nil
+	rep := &stats{Interval: int(s.t.balanceTime.Sub(s.t.addTime).Seconds())}
```
Comment (lhy1024) [inline bench.go:135]: would you like to show the report example?

Comment (ZenoTan) [reply bench.go:135]: `Prev Balance leader is 50746.00, compared to origin by 0.00
Prev balance region is 63512.00, compared to origin by 0.00
Cur balance leader is 50746.00, compared to origin by 0.00
Cur balance region is 63512.00, compared to origin by 0.00
Prev latency is 0.00, compared to origin by 0.00
Cur latency is 0.00, compared to origin by 0.00`

File: bench.go (hunk 1 above)
Comment (ZenoTan) [reply bench.go:11]: The latest offical prometheus used this as an example...

File: bench.go (hunk 2 above)
Comment (ZenoTan) [reply bench.go:25]: Collect it by count of schedules in balance region scheduler.

**Pull request merged**